### DISCUSSION TITLE
Path parameter encoding in client and controller

### DISCUSSION THREAD

**Discussion by goeh** [micronaut-projects/micronaut-core#3967] [Q&A] [answered]
Hi, I'm developing a file browser type of controller and I'm confused about path encoding when using {+path} syntax in the controller and client. I made a test application that exercises a few request variants.

https://github.com/goeh/micronaut-url-encoding-issue

```java
@Client("/")
public interface TestClient {

    @Get("{path}")
    String standard(String path);

    @Get("{+path}")
    String includeReservedChars(String path);
}

@Controller("/")
public class TestController {
    @Get(value = "/{+path}", produces = MediaType.TEXT_PLAIN)
    public String test(String path) {
        return path;
    }
}
```

Most of the test results makes sense but I was surprised that character encoding got disabled/wrong when using {+path} on the declarative client. For example "/foo/bar/göran.txt" results in "/foo/bar/gÃ¶ran.txt".

My findings resulted in that I should not use {+path} on the client side, only on the server/controller side. Is Micronaut doing everything correct here? I'm using Micronaut 1.3.7

**Comment by graemerocher** [comment, marked as answer]
Could be a bug, be the encoding used to parse the path parameters. Maybe worthy of an issue report

**Reply by goeh** [reply]
Thanks. I filed an issue https://github.com/micronaut-projects/micronaut-core/issues/3974
